app/upgrades/v13/upgrades.go

```diff
@@ -1,15 +1,84 @@
 package v13
 
 import (
+	"fmt"
+	"os"
+	"strings"
+
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/types/module"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
 	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
-
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
 	"github.com/osmosis-labs/osmosis/v12/app/keepers"
 	"github.com/osmosis-labs/osmosis/v12/app/upgrades"
+	ibcratelimittypes "github.com/osmosis-labs/osmosis/v12/x/ibc-rate-limit/types"
 	lockuptypes "github.com/osmosis-labs/osmosis/v12/x/lockup/types"
 )
 
+func setupRateLimiting(ctx sdk.Context, keepers *keepers.AppKeepers) error {
+	govModule := keepers.AccountKeeper.GetModuleAddress(govtypes.ModuleName)
+	code, err := os.ReadFile("rate_limiter.wasm")
+	if err != nil {
+		return err
+	}
+	contractKeeper := wasmkeeper.NewGovPermissionKeeper(keepers.WasmKeeper)
+	instantiateConfig := wasmtypes.AccessConfig{Permission: wasmtypes.AccessTypeOnlyAddress, Address: govModule.String()}
+	codeID, err := contractKeeper.Create(ctx, govModule, code, &instantiateConfig)
+	if err != nil {
+		return err
+	}
+
+	transferModule := keepers.AccountKeeper.GetModuleAddress(transfertypes.ModuleName)
+
+	denoms := []string{
+		"uosmo",
+		"ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2", // atom
+		"ibc/D189335C6E4A68B513C10AB227BF1C1D38C746766278BA3EEB4FB14124F1D858", // usdc
+		"ibc/EA1D43981D5C9A1C4AAEA9C23BB1D4FA126BA9BC7020A25E0AE4AA841EA25DC5", // weth
+		"ibc/EA1D43981D5C9A1C4AAEA9C23BB1D4FA126BA9BC7020A25E0AE4AA841EA25DC5", // wbtc
```

Comment:
I think this is the right one but please double-check

```suggestion
		"ibc/D1542AA8762DB13087D8364F3EA6509FD6F009A34F00426AF9E4F9FA85CBBF1F", // wbtc
```